src/app/user-profile/user-profile.component.spec.ts

```diff
@@ -19,9 +47,25 @@ describe('UserProfileComponent', () => {
     fixture = TestBed.createComponent(UserProfileComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    userServiceSpy = fixture.debugElement.injector.get(UserService);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('checkIsValidUser true on valid user', () => {
+    expect(component.checkIsValidUser(mockUser)).toBeTrue();
+  });
+
+  it('checkUser should set the user', () => {
+    component.setUser(mockUser);
+    expect((component.user as User).firstName).toEqual('Foo');
```

Comment:
this only checks the `firstName` is being set correctly. It should be fleshed out a bit to include the rest of the fields. Same for the below with addr